File: domain/src/main/java/project/price_it/entity/UserEntity.java
package project.price_it.entity;

import jakarta.persistence.*;
import lombok.*;
import project.price_it.entity.point.PointEntity;
import project.price_it.entity.point.PointType;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Builder
public class UserEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String email;
    @Column
    private String password;

    @Column
    private String name;

    @Column
    private int point;

    @Column
    private Double lat;
    @Column
    private Double lng;

    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PointEntity> points = new ArrayList<>();
}

File: domain/src/main/java/project/price_it/entity/UserEntity.kt
package project.price_it.entity

import jakarta.persistence.*
import lombok.Builder
import lombok.Getter
import lombok.Setter
import lombok.ToString

@Entity
@ToString
@Builder
@Getter
@Setter
@Table(name = "user_entity")
class UserEntity(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @Column
        var email: String,

        @Column
        var password: String,

        @Column
        var name: String,

        @Column
        var point: Int = 0,

        @Column
        var lat: Double? = null,

        @Column
        var lng: Double? = null
)
